engine/modules/opentelemetry/src/opentelemetry.cc

```diff
@@ -0,0 +1,279 @@
+    //       host_serv_metric hostservmetric =
+    //           last_success->second->extract_host_serv_metric(data_pt);
+
+    //       if (!hostservmetric.host.empty()) {
+    //         _fifo.add_data_point(hostservmetric.host, hostservmetric.service,
+    //                              hostservmetric.metric, data_pt);
+
+    //         auto waiting = host_serv_index.equal_range(
+    //             host_serv{hostservmetric.host, hostservmetric.service});
+    //         while (waiting.first != waiting.second) {
+    //           to_notify.push_back(*waiting.first);
+    //           waiting.first = host_serv_index.erase(waiting.first);
+    //         }
+    //         data_point_known = true;
+    //         break;
+    //       }
+    //       ++last_success;
+    //       if (last_success == _extractors.end()) {
+    //         last_success = _extractors.begin();
+    //       }
+    //     }
+    //     if (!data_point_known) {
+    //       unknown.push_back(data_pt);  // unknown metric => forward to broker
+    //     }
+    //   });
+    //   SPDLOG_LOGGER_TRACE(_logger, "fifos:{}", _fifo);
+    //   // we wait that all request datas have been computed to give us more
+    //   // chance of converter success
+    //   for (auto to_callback : to_notify) {
+    //     if (!to_callback->async_build_result_from_metrics(
+    //             _fifo)) {  // not enough data => repush in _waiting
+    //       _waiting.insert(to_callback);
+    //     }
+    //   }
+    //   SPDLOG_LOGGER_TRACE(_logger, "fifos:{}", _fifo);
+    // }
+  }
+  if (!unknown.empty()) {
+    SPDLOG_LOGGER_TRACE(_logger, "{} unknown data_points", unknown.size());
+    _forward_to_broker(unknown);
+  }
+}
+
+/**
+ * @brief the second timer is used to handle converter timeouts
+ *
+ */
+void open_telemetry::_start_second_timer() {
+  std::lock_guard l(_protect);
+  _second_timer.expires_from_now(std::chrono::seconds(1));
+  _second_timer.async_wait(
+      [me = shared_from_this()](const boost::system::error_code& err) {
+        if (!err) {
+          me->_second_timer_handler();
+        }
+      });
+}
+
+/**
+ * @brief notify all timeouts
+ *
+ */
+void open_telemetry::_second_timer_handler() {
+  //  std::vector<std::shared_ptr<otl_converter>> to_notify;
+  {
+    std::lock_guard l(_protect);
+    std::chrono::system_clock::time_point now =
+        std::chrono::system_clock::now();
+    // waiting_converter::nth_index<1>::type& expiry_index = _waiting.get<1>();
+    // while (!_waiting.empty()) {
+    //   auto oldest = expiry_index.begin();
+    //   if ((*oldest)->get_time_out() > now) {
+    //     break;
+    //   }
+    //   to_notify.push_back(*oldest);
+    //   expiry_index.erase(oldest);
+    // }
+  }
+
+  // notify all timeout
+  // for (std::shared_ptr<otl_converter> to_not : to_notify) {
+  //   SPDLOG_LOGGER_DEBUG(_logger, "time out: {}", *to_not);
+  //   to_not->async_time_out();
+  // }
+
+  _start_second_timer();
```

Comment:
I have the feeling that, this is not useful at the moment.

Reply:
next PR